test-builder/elements.ts

```diff
@@ -131,14 +131,48 @@ const build = async (specElements, customElements) => {
             true,
           );
 
-          const defaultAttrCode = `(function() {
-    var instance = ${defaultConstructCode};
-    return !!instance && '${attrProp}' in instance;
-  })()`;
+          let attrCode = `(function() {
+            var instance = ${defaultConstructCode};
+            return !!instance && '${attrProp}' in instance;
+          })()`;
+
+          // SVG attributes need some tweaks
+          if (category === "svg") {
+            // Certain SVG attributes are reflected differently in SVGOM
+            if (attrProp === "in") {
+              attrCode = attrCode.replace("'in'", "'in1'");
+            }
+            if (attrProp === "kernelUnitLength") {
+              attrCode = attrCode.replace(
+                "kernelUnitLength",
+                "kernelUnitLengthX",
+              );
+            }
+            if (attrProp === "order") {
+              attrCode = attrCode.replace("order", "orderX");
+            }
+            if (attrProp === "stdDeviation") {
+              attrCode = attrCode.replace("stdDeviation", "stdDeviationX");
+            }
+            if (attrProp === "radius") {
+              attrCode = attrCode.replace("radius", "radiusX");
+            }
+            if (attrProp === "baseFrequency") {
+              attrCode = attrCode.replace("baseFrequency", "baseFrequencyX");
+            }
+            // All xlink:href attributes need special handling
+            if (attrProp === "xlink_href") {
+              attrCode = `(function() {
+                var instance = ${defaultConstructCode};
+                instance.setAttributeNS('http://www.w3.org/1999/xlink', 'xlink:href', 'test');
+                return !!instance && instance.getAttributeNS('http://www.w3.org/1999/xlink', 'href') === 'test'
+              })()`;
+            }
+          }
```

Comment:
I'm not sure if we want to do this special handling here in the test builder -- maybe adding custom tests would be more suitable?

Reply:
I did that originally in https://github.com/openwebdocs/mdn-bcd-collector/pull/990/commits/a35280cb520e39664932c71e37b4b0ba167bd2d8 just for Xlink:href. The other attributes are also used many times for various SVG elements. So, this will lead to many custom tests. If you prefer, I can write all of them. Just a heads up that it will be a lot...

Reply:
Okay, if it's going to be a significant amount of custom tests to write, then this would probably be the better option!

For simplification and ease of future maintenance, perhaps we could do the following instead?

```js
const replacements = {
  "in": "in1",
  "kernelUnitLength": "kernelUnitLengthX",
  // ...
};

if (attrProp in replacements) {
  attrCode = attrCode.replace(`'${attrProp}'`, `'${replacements[attrProp]}'`);
}

// All xlink attributes need special handling
if (attrProp.startsWith("xlink_")) {
  const xlinkAttr = attrProp.replace("xlink_", "");
  attrCode = `(function() {
    var instance = ${defaultConstructCode};
    instance.setAttributeNS('http://www.w3.org/1999/xlink', 'xlink:${xlinkAttr}', 'test');
    return !!instance && instance.getAttributeNS('http://www.w3.org/1999/xlink', '${xlinkAttr}') === 'test'
  })()`;
}
```

Reply:
Thanks, this is nicer! 🎉